### PR TITLE
appgw: validate cert only if certificateFile is specified in set ssl cert command

### PR DESCRIPTION
**Pull request by akshaysngupta**
## Description
This PR majorly is to fix https://github.com/Azure/azure-powershell/issues/9488

But as a part of this PR, we add a test in Network project using a KeyVault dataplane cmdlets. To make these cmdlets work, I am have made additional changes to Network Test Runner and KeyVault project. These changes help to mock the keyvault client in the Network project.

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Network/Network/ApplicationGateway/SslCertificate/SetAzureApplicationGatewaySslCertificateCommand.cs [hunk 1]
```diff
@@ -40,7 +40,9 @@ public override void ExecuteCmdlet()
                 throw new ArgumentException("Ssl certificate with the specified name does not exist");
             }
             
-            X509Certificate2 cert = new X509Certificate2(this.CertificateFile, this.Password, X509KeyStorageFlags.Exportable);
+            if (this.CertificateFile != null) {
```
Comment (markcowl) [inline src/Network/Network/ApplicationGateway/SslCertificate/SetAzureApplicationGatewaySslCertificateCommand.cs]: We need a test to protect us from regressing after this fix goes in.  Also, need a change in the network changelog, per the PR checklist

**Comment by VeryEarly** [comment]
Hi @akshaysngupta ,

Could you please also update the changelog.md?

File: src/Network/Network/ApplicationGateway/SslCertificate/SetAzureApplicationGatewaySslCertificateCommand.cs (hunk 1 above)
Comment (akshaysngupta) [reply src/Network/Network/ApplicationGateway/SslCertificate/SetAzureApplicationGatewaySslCertificateCommand.cs]: @markcowl Do we have the ability to mock keyvault responses now for data plane ?

Comment (markcowl) [reply src/Network/Network/ApplicationGateway/SslCertificate/SetAzureApplicationGatewaySslCertificateCommand.cs]: No, but we can mock the keyvault client altogether. there should at least be a test so we can validate the network portion

**Comment by number213** [comment]
Hi @akshaysngupta, changes from `network-march` were merged into master. Please retarget your PR to master as well to make the release. Please also note that `master` was rebased due to credentials leaking into it so it's basically incompatible with `network-march`. My suggestion is to create a new branch from master and copy your changes there.

**Comment by number213** [comment]
@akshaysngupta please re-create PR using `master` as base

**Comment by RustyF** [comment]
Any idea what release this will be in?

**Pull request closed without merging**